cardano-cli/test/cardano-cli-golden/Test/Golden/ErrorsSpec.hs

```diff
@@ -0,0 +1,97 @@
+    , ("VotingCredentialDecodeGovCmdEror"
+      , VotingCredentialDecodeGovCmdEror $ DecoderErrorEmptyList "emptylist")
+    , ("WriteFileError"
+      , WriteFileError $ FileError "path/file.txt" ())
+    , ("ReadFileError"
+      , ReadFileError $ FileError "path/file.txt" InputInvalidError)
+    , ("ExpectedStakeKeyCredentialGovCmdError"
+      , ExpectedStakeKeyCredentialGovCmdError)
+    , ("NonUtf8EncodedConstitution"
+      , NonUtf8EncodedConstitution $ DecodeError "seq" Nothing)
+    , ("GovernanceCmdTextEnvReadError"
+      , GovernanceCmdTextEnvReadError
+        . FileError  "path/file.txt"
+        $ TextEnvelopeAesonDecodeError "cannot decode json")
+    , ("GovernanceCmdCddlError"
+      , GovernanceCmdCddlError
+        $ CddlErrorTextEnv
+          (FileError "path/file.txt" . TextEnvelopeDecodeError $ DecoderErrorCustom "todecode" "decodeerr")
+          (FileError "path/file.txt" TextEnvelopeCddlUnknownKeyWitness))
+    , ("GovernanceCmdKeyReadError"
+      , GovernanceCmdKeyReadError $ FileError "path/file.txt" InputInvalidError)
+    , ("GovernanceCmdCostModelReadError"
+      , GovernanceCmdCostModelReadError $ FileError "path/file.txt" ())
+    , ("GovernanceCmdTextEnvWriteError"
+      , GovernanceCmdTextEnvWriteError $ FileError "path/file.txt" ())
+    , ("GovernanceCmdEmptyUpdateProposalError"
+      , GovernanceCmdEmptyUpdateProposalError)
+    , ("GovernanceCmdMIRCertificateKeyRewardMistmach"
+      , GovernanceCmdMIRCertificateKeyRewardMistmach "path/file.txt" 1 2)
+    , ("GovernanceCmdCostModelsJsonDecodeErr"
+      , GovernanceCmdCostModelsJsonDecodeErr "path/file.txt" "jsonerr")
+    , ("GovernanceCmdEmptyCostModel"
+      , GovernanceCmdEmptyCostModel "path/file.txt")
+    , ("GovernanceCmdUnexpectedKeyType"
+      , GovernanceCmdUnexpectedKeyType (TextEnvelopeType <$> ["env1", "env2"]))
+    , ("GovernanceCmdPollOutOfBoundAnswer"
+      , GovernanceCmdPollOutOfBoundAnswer 4)
+    , ("GovernanceCmdPollInvalidChoice"
+      , GovernanceCmdPollInvalidChoice)
+    , ("GovernanceCmdDecoderError"
+      , GovernanceCmdDecoderError $ DecoderErrorEmptyList "empty")
+    , ("GovernanceCmdVerifyPollError"
+      , GovernanceCmdVerifyPollError ErrGovernancePollNoAnswer)
+    , ("GovernanceCmdWriteFileError"
+      , GovernanceCmdWriteFileError $ FileError "path/file.txt" ())
+    ]
+
+
+goldenFilesPath :: FilePath
+goldenFilesPath = "test/cardano-cli-golden/files/golden/errors"
+
+_testAllErrorMessages :: forall a. (HasCallStack, Data a, Error a) => [a] -> TestTree
+_testAllErrorMessages = ErrorMessage.testAllErrorMessages goldenFilesPath
+
+testAllErrorMessages_ :: forall a. (HasCallStack, Error a)
+                      => String -- ^ module name
+                      -> String -- ^ type name
+                      -> [(String, a)]  -- ^ list of constructor names and values
+                      -> TestTree
+testAllErrorMessages_ = ErrorMessage.testAllErrorMessages_ goldenFilesPath
```

Comment:
If you are going to do this refactor (which I think it's not worth it) then give the function a better name.

Reply:
Do you mean by refactor, the change in cardano-api? I wanted to avoid copying code.

I've renamed the function.